addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/handler/MiHomeSubdevicesHandler.java

```diff
@@ -0,0 +1,651 @@
+                        break;
+                }
+                break;
+            case OFFLINE:
+
+            case UNKNOWN:
+
+            case UNINITIALIZED:
+                if (getThing().getStatus() == ThingStatus.ONLINE) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, bridgeStatus.toString());
+                }
+                break;
+            case REMOVED:
+                logger.warn(
+                        "Gateway has been removed, subdevice {} will be removed as it has been deleted from the MiHome server.",
+                        mihomeID);
+                updateStatus(ThingStatus.REMOVED);
+                break;
+            default:
+                // No action needed, we keep the current thing status
+                break;
+        }
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        // Validate against the config description
+        validateConfigurationParameters(configurationParameters);
+
+        // Update the configuration
+        Configuration configuration = editConfiguration();
+        BigDecimal interval = (BigDecimal) configurationParameters.get(MiHomeBindingConstants.CONFIG_UPDATE_ITNERVAL);
+        configuration.put(MiHomeBindingConstants.CONFIG_UPDATE_ITNERVAL, interval);
+        updateConfiguration(configuration);
+
+        this.updateInterval = interval.longValue();
+
+        // Reschedule the job if it is already started
+        scheduleRegularUpdate(this.updateInterval);
+    }
+
+    private JsonObject getNewDevice(JsonArray oldDevices, JsonArray newDevices, String deviceType, int gatewayID) {
+        Set<Integer> oldIDs = new HashSet<Integer>();
+        // Save the IDs of the old devices that are of the searched type for this bridge
+        for (JsonElement device : oldDevices) {
+            JsonObject deviceObj = (JsonObject) device;
+            String type = deviceObj.get(DeviceConstants.DEVICE_TYPE_KEY).getAsString();
+            int gateway = deviceObj.get(DeviceConstants.SUBDEVICE_PARENT_ID_KEY).getAsInt();
+
+            if (type.equals(deviceType) && gateway == gatewayID) {
+                int deviceID = deviceObj.get(DeviceConstants.DEVICE_ID_KEY).getAsInt();
+                oldIDs.add(deviceID);
+            }
+        }
+
+        // Search for new devices
+        for (JsonElement device : newDevices) {
+            JsonObject deviceObj = (JsonObject) device;
+            String type = deviceObj.get(DeviceConstants.DEVICE_TYPE_KEY).getAsString();
+            int deviceID = deviceObj.get(DeviceConstants.DEVICE_ID_KEY).getAsInt();
+            int gateway = deviceObj.get(DeviceConstants.SUBDEVICE_PARENT_ID_KEY).getAsInt();
+
+            boolean isNewDeviceFound = type.equals(deviceType) && gateway == gatewayID && !oldIDs.contains(deviceID);
+            if (isNewDeviceFound) {
+                return deviceObj;
```

Comment:
> This only returns the first new device, why not return them all? 

The instance of ThingHandler is responsible for a single unique device, you can't have a single Handler for multiple devices. What I am supposed to do with the other devices ?

> Note that the comment (line 424) suggests that searches for all new devices

You have right, I will change the line and maybe add additional comments, as the case is very specific.